Easydict/Swift/Service/OpenAI/BaseOpenAIService.swift

```diff
@@ -0,0 +1,251 @@
+        .openAI
+    }
+
+    override public func name() -> String {
+        NSLocalizedString("openai_translate", comment: "")
+    }
+
+    override public func link() -> String? {
+        "https://chat.openai.com"
+    }
+
+    override public func queryTextType() -> EZQueryTextType {
+        var type: EZQueryTextType = []
+        let enableTranslation = UserDefaults.standard.string(forKey: EZOpenAITranslationKey) ?? "0"
+        if enableTranslation != "0" {
+            type.insert(.translation)
+        }
+
+        let enableDictionary = UserDefaults.standard.string(forKey: EZOpenAIDictionaryKey) ?? "0"
+        if enableDictionary != "0" {
+            type.insert(.dictionary)
+        }
+
+        let enableSentence = UserDefaults.standard.string(forKey: EZOpenAISentenceKey) ?? "0"
+        if enableSentence != "0" {
+            type.insert(.sentence)
+        }
+
+        return type
+    }
+
+    override public func intelligentQueryTextType() -> EZQueryTextType {
+        Configuration.shared.intelligentQueryTextTypeForServiceType(serviceType())
+    }
+
+    override public func supportLanguagesDictionary() -> MMOrderedDictionary<AnyObject, AnyObject> {
+        let orderedDict = MMOrderedDictionary<AnyObject, AnyObject>()
+        for language in EZLanguageManager.shared().allLanguages {
+            var value = language.rawValue
+            if language == .classicalChinese {
+                value = Language.wenyanwen
+            }
+
+            if language != .burmese {
+                orderedDict.setObject(value as NSString, forKey: language.rawValue as NSString)
+            }
+        }
+
+        return orderedDict
+    }
+
+    // swiftlint:disable identifier_name
+    override public func translate(
+        _ text: String,
+        from: Language,
+        to: Language,
+        completion: @escaping (EZQueryResult, Error?) -> ()
+    ) {
+        let url = URL(string: endPoint)
+        let invalidURLError = EZError(type: .param, description: "\(serviceType().rawValue) URL is invalid")
+        guard let url, url.isValid else { completion(result, invalidURLError); return }
+
+        var resultText = ""
+
+        result.from = from
+        result.to = to
+
+        let queryType = queryTextType(text: text, from: from, to: to)
+        let chats = chatMessages(queryType: queryType, text: text, from: from, to: to)
+        let query = ChatQuery(messages: chats, model: model, temperature: 0)
+        let openAI = OpenAI(apiToken: apiKey)
+
+        openAI.chatsStream(query: query, url: url) { [weak self] res in
+            guard let self else { return }
+
+            switch res {
+            case let .success(chatResult):
+                if let content = chatResult.choices.first?.delta.content {
+                    resultText += content
+                    handleResult(queryType: queryType, resultText: resultText, error: nil, completion: completion)
+                }
```

Comment:
Do we need to handle the else case here?

Reply:
I am not sure, but I handled it just in case.